### PR TITLE
[ready] network multilabel

### PR DESCRIPTION
**Pull request by sbma44**
refs https://github.com/ingalls/pt2itp/issues/180

### CODE REVIEW

File: test/cluster.clusterWithin.test.js [hunk 1]
```diff
@@ -134,8 +134,8 @@ test('LinesStrings far away should not be clustered', (t) => {
         pool.query(`
             BEGIN;
             DROP TABLE IF EXISTS network;
-            CREATE TABLE network (id SERIAL, segment BIGINT, text TEXT, text_tokenless TEXT, _text TEXT, named BOOLEAN, geom GEOMETRY(LINESTRING, 4326));
-            CREATE TABLE network_cluster(ID SERIAL, text TEXT, _text TEXT, text_tokenless TEXT, geom GEOMETRY(GEOMETRY, 4326), buffer GEOMETRY(Polygon,4326), address INTEGER);
+            CREATE TABLE network (id SERIAL, segment BIGINT, text TEXT, text_tokenless TEXT, _text TEXT, named BOOLEAN, geom GEOMETRY(LINESTRINGZ, 4326));
```
Comment (boblannon) [inline test/cluster.clusterWithin.test.js]: the type for `id` should be changed to `BIGINT`, to match what's going on in `copy.js` now

**Comment by boblannon** [comment]
got this to run on US MT.  here are the side-by-side numbers before and after the change:

```
| ERROR TYPE             | BEFORE | AFTER  |
|------------------------|--------|--------|
| DIST                   | 21     | 22     |
| NAME MISMATCH (HARD)   | 864    | 2342   |
| NAME MISMATCH (SOFT)   | 404    | 374    |
| NOT MATCHED TO NETWORK | 108075 | 101532 |
| TEXT                   | 259    | 137    |
```

I'll dig more into those tomorrow but `NOT MATCHED TO NETWORK` has gone down by thousands.  `NAME MISMATCH (HARD)` seems to have increased, though, so i'll check into that.

**Comment by sbma44** [comment]
Hmm. I was hoping for a much bigger improvement than this.

**Comment by boblannon** [comment]
yeah, me too.  my back-of-the-envelope estimation was around 38k addresses that had something like "highway" in their names.

**Comment by boblannon** [comment]
the endcap param for ST_Buffer might be useful when trying to find overlap between two conterminous linestrings: 

2017-09-29 at 10 33 am

**Comment by sbma44** [comment]
@boblannon ah but I'm preserving the unique IDs of each geometry fragment -- no geometric operations are necessary to determine overlap, just some set operations.

**Comment by sbma44** [comment]
(but I should add that @ingalls does use techniques just like this in other parts of the codebase when relating network clusters to address clusters)

**Comment by sbma44** [comment]
OK! This is ready for a review & merge. **BUT FIRST WE SHOULD CONFIRM THAT IT ACTUALLY HELPS.** I feel good about the unit tests in place; I feel good about the need for this code; I feel good about the implementation. But the proof is in the passrate.

@ingalls, would love a review.

**Comment by ingalls** [comment]
I've tested this out on several different sources now and not seeing any alarms. PT2ITP test mode shows imporved passrates. Going to go forward with a merge here. Thanks a ton for all your hard work @sbma44!

**Comment by ingalls** [comment]
:taco: :tada:

**Pull request merged**